packages/itwinui-react/src/core/utils/hooks/useTheme.ts

```diff
@@ -59,6 +59,14 @@ export const useTheme = (
       return;
     }
 
+    // do nothing if using v1 for the main page (incremental migration)
+    if (
+      ownerDocument.documentElement.className.includes('iui-theme-') ||
+      ownerDocument.body.classList.contains('iui-body')
+    ) {
+      return;
+    }
+
     ownerDocument.body.classList.toggle('iui-root', true);
```

Comment:
If I understand this correctly, the addition of `iui-root` here and/or the rest of the code is what is breaking v1 style in the presence of v1 classes? Why is that the case?

Reply:
Yes, it is both. Here's how it works:
1. in v1, when using a component directly, the `useTheme` hook will add `iui-body` to `<body>` and `iui-theme-*` classes to the document element (`<html>`).
  ![](https://user-images.githubusercontent.com/9084735/224784972-fd0eb983-eb6f-4608-bea4-9945de5eb5c5.png)
2. in v2, when using a component directly, the `useTheme` hook will add `iui-root` to `<body>` and `data-iui-*` attributes to `<html>`.
  ![](https://user-images.githubusercontent.com/9084735/224784188-4d4e947e-0bc4-438f-807f-46e2bfbdf489.png) 
3. in v2, when wrapping the component tree in `<ThemeProvider>`, the `useTheme` hook will do nothing because `<ThemeProvider>` automatically adds those attributes/classses. This ensures that nothing outside ThemeProvider is affected (it is considered "scoped").

For this bug specifically, instead of 1 and 3 in the above list, 1 and 2 are happening. 1 and 3 are compatible but 1 and 2 are not.

Does that help?

Reply:
Yes that makes sense. Thanks for explaining 😸